gateway/src/http.ts

```diff
@@ -14,8 +14,14 @@ export const internalRedirect = async (
 
   const response = await fetch(_url, {
     headers,
+    signal: AbortSignal.timeout(30_000),
```

Comment:
Is 30 seconds sufficient? Should this be configurable?

Reply:
I agree should be configurable